src/lib/studio/studio.svelte

```diff
@@ -5,19 +5,56 @@
 <script lang="ts">
     import './shim.css';
     import { onMount } from 'svelte';
-    import { ensureStudioComponent, initImagine, getWebComponents } from './studio-widget';
+    import { resolve } from '$app/paths';
+    import { Link } from '$lib/elements';
     import { app } from '$lib/stores/app';
-    import { invalidate } from '$app/navigation';
     import { Dependencies } from '$lib/constants';
+    import { goto, invalidate } from '$app/navigation';
+    import { IconExternalLink } from '@appwrite.io/pink-icons-svelte';
+    import { Layout, Typography, Icon } from '@appwrite.io/pink-svelte';
+    import { ensureStudioComponent, initImagine, getWebComponents } from './studio-widget';
+    import DomainsTable from './domainsTable.svelte';
+    import SideSheet from '$routes/(console)/project-[region]-[project]/databases/database-[database]/table-[table]/layout/sidesheet.svelte';
+
+    const {
+        region,
+        projectId
+    }: {
+        region: string;
+        projectId: string;
+    } = $props();
 
-    const { region, projectId }: { region: string; projectId: string } = $props();
+    const siteId = `project-${projectId}`;
+    let showManageDomainsSheet = $state(false);
+    let primaryDomainForSite = $state(`imagine-${projectId}.stage.appwrite.network`);
+
+    const addDomainUrl = $derived.by(() => {
+        const baseUrl = resolve(
+            '/(console)/project-[region]-[project]/sites/site-[site]/domains/add-domain',
+            {
+                region,
+                project: projectId,
+                site: siteId
+            }
+        );
+        return `${baseUrl}?types=false`;
+    });
 
     onMount(() => {
         ensureStudioComponent();
 
         initImagine(region, projectId, {
             onProjectNameChange: () => {
                 invalidate(Dependencies.PROJECT);
+            },
+            onAddDomain: async () => {
+                await goto(addDomainUrl);
```

Comment:
since this is only used here, we dont have to derive it right? we can just have the derivation logic in the callback